daemon/stats/collector_unix.go

```diff
@@ -54,15 +64,21 @@ func (s *Collector) getSystemCPUUsage() (uint64, error) {
 			for _, i := range parts[1:8] {
 				v, err := strconv.ParseUint(i, 10, 64)
 				if err != nil {
-					return 0, fmt.Errorf("Unable to convert value %s to int: %s", i, err)
+					return 0, fmt.Errorf("unable to convert value %s to int: %s", i, err)
 				}
 				totalClockTicks += v
+
 			}
-			return (totalClockTicks * nanoSecondsPerSecond) /
-				clockTicksPerSecond, nil
+
+			if totalClockTicks > maxClockTicks {
+				return 0, fmt.Errorf("number of clock ticks obtained from '%s' will overflow cpu stats calculations", procStatFile.Name())
```

Comment:
Haven't looked too closely, but wondering if it's still possible (and if it would make sense) to return something usable in this case (instead of erroring out).

Reply:
Thinking about it more in depth, these values are all relative to each other so the calculations for CPU should be more or less OK after giving out wrong information once.

That is to say, since the values are compared against each other, they'll be overflowing relative to each other https://play.golang.org/p/w1RnkoONL6g

The only caveat is that a single report will be totally incorrect because of the overflow. I think that's an OK compromise and we can get rid of this check, and just leave a comment saying the above.

Reply:
I should add, this is highly dependent on the calculations done in https://github.com/docker/cli/blob/c6bb56136f20ed969dcad271b9167b46941ae1f0/cli/command/container/stats_helpers.go#L166 continuing to be done the same way as they are done right now. If the way those calculations are ever changed it this may not hold up.